test/jdk/java/awt/geom/Path2D/GetBounds2DPrecisionTest.java

```diff
@@ -0,0 +1,259 @@
+            } else if (k == PathIterator.SEG_CLOSE) {
+                returnValue.append("z");
+            }
+            pi.next();
+        }
+        return returnValue.toString();
+    }
+
+    private static String toUniformString(double value) {
+        BigDecimal decimal = new BigDecimal(value);
+        int DIGIT_COUNT = 40;
+        String str = decimal.toPlainString();
+        if (str.length() >= DIGIT_COUNT) {
+            str = str.substring(0,DIGIT_COUNT-1)+"…";
+        }
+        while(str.length() < DIGIT_COUNT) {
+            str = str + " ";
+        }
+        return str;
+    }
+
+    private static String toComparisonString(String target, String observed) {
+        for(int a = 0; a<target.length(); a++) {
+            char ch1 = target.charAt(a);
+            char ch2 = observed.charAt(a);
+            if (ch1 != ch2) {
+                return observed.substring(0,a) + createCircleDigit(ch2)+observed.substring(a+1);
+            }
+        }
+        return observed;
+    }
+
+    /**
+     * Convert a digit 0-9 into a "circle digit". Really we just want any unobtrusive way to
+     * highlight a character.
+     */
+    private static char createCircleDigit(char ch) {
+        if (ch >= '1' && ch <='9')
+            return (char)( ch - '1' + '\u2460');
+        if (ch == '0')
+            return '\u24ea';
+        return ch;
+    }
+
+    private static int solveQuadratic(BigDecimal[] eqn, BigDecimal[] res) {
+        BigDecimal a = eqn[2];
+        BigDecimal b = eqn[1];
+        BigDecimal c = eqn[0];
+        int roots = 0;
+        if (a.equals(BigDecimal.ZERO)) {
+            // The quadratic parabola has degenerated to a line.
+            if (b.equals(BigDecimal.ZERO)) {
+                // The line has degenerated to a constant.
+                return -1;
+            }
+            res[roots++] = c.negate().divide(b);
+        } else {
+            // From Numerical Recipes, 5.6, Quadratic and Cubic Equations
+            BigDecimal d = b.multiply(b).add(new BigDecimal(-4.0).multiply(a).multiply(c));
+            if (d.compareTo(BigDecimal.ZERO) < 0) {
+                // If d < 0.0, then there are no roots
+                return 0;
+            }
+            d = d.sqrt(MathContext.DECIMAL128);
+            // For accuracy, calculate one root using:
+            //     (-b +/- d) / 2a
+            // and the other using:
+            //     2c / (-b +/- d)
+            // Choose the sign of the +/- so that b+d gets larger in magnitude
+            if (b.compareTo(BigDecimal.ZERO) < 0) {
+                d = d.negate();
+            }
+            BigDecimal q = b.add(d).divide(new BigDecimal(-2.0));
+            // We already tested a for being 0 above
+            res[roots++] = q.divide(a, RoundingMode.HALF_EVEN);
+            if (!q.equals(BigDecimal.ZERO)) {
+                res[roots++] = c.divide(q, RoundingMode.HALF_EVEN);
```

Comment:
This divide operation suffers missing decimals:
Use BigDecomal.setScale(40) to ascertain precision.

See https://github.com/bourgesl/marlin-math/blob/c9bbfbac4565e81e54ce79fc02c38fcaa2fe0e48/src/main/java/test/FindExtremaAccuracyTest.java#L667

See cubic test fixed:
https://github.com/bourgesl/marlin-math/blob/main/results/findExtrema/jdk17-2021-11-19-dist-cubic.log